packages/app/src/app/pages/Dashboard/Components/V2Repository/index.tsx

```diff
@@ -0,0 +1,156 @@
+import React from 'react';
+import {
+  Element,
+  Column,
+  Grid,
+  Icon,
+  IconButton,
+  ListAction,
+  Stack,
+  Text,
+} from '@codesandbox/components';
+import { css } from '@styled-system/css';
+
+import { useAppState } from 'app/overmind';
+import { DashboardV2Repository } from '../../types';
+
+export const V2Repository: React.FC<DashboardV2Repository> = ({ repo }) => {
+  const { branches, repository } = repo;
+  const {
+    dashboard: { viewMode },
+  } = useAppState();
+  const ariaLabel = `View branches from repository ${repository.name} by ${repository.owner}`;
+  const branchesLabel = `${branches.length} ${
+    branches.length === 1 ? 'branch' : 'branches'
+  }`;
+  // TODO: handle selecting a repository
+  const handleClick = (_: React.MouseEvent | React.KeyboardEvent) => ({});
+
+  if (viewMode === 'grid') {
+    return (
+      <Stack
+        aria-label={ariaLabel}
+        css={css({
+          cursor: 'pointer', // TODO: revisit cursor.
+          position: 'relative',
+          overflow: 'hidden',
+          height: 240,
+          width: '100%',
+          borderRadius: '4px',
+          background: '#191919',
+          transition: 'background ease-in-out',
+          padding: 6,
+
+          transitionDuration: theme => theme.speeds[4],
+          ':hover, :focus, :focus-within': {
+            // This is not the official transition.
+            backgroundColor: '#252525',
+          },
+          ':focus-visible': {
+            boxShadow: '0 0 2px 1px rgba(255, 255, 255, 0.4)',
+          },
+        })}
+        direction="vertical"
+        gap={4}
+        onClick={handleClick}
+        onKeyDown={e => e.key === 'Enter' && handleClick(e)}
+        // TODO: refine semantics when/if the context menu gets implemented.
+        role="link"
```

Comment:
let's do a quick chat with the design team about the interaction model in the dashboard. `branches` and `repos` are links, while the sandboxes are selectable via one click and are open by dbl click. wonder if we should mimic that or disable selection for the pages where we show branch and repo cards

Reply:
Will be addressed in https://linear.app/codesandbox/issue/XTD-140/recent-page-improve-interaction-model.